spec/prawn/images_spec.rb

```diff
@@ -43,11 +43,68 @@
     expect(info.height).to eq(453)
   end
 
+  it 'does not close passed-in IO objects' do
+    file = File.open(filename, 'rb')
+    _info = pdf.image(file)
+
+    expect(file).to_not be_closed
+  end
+
   it 'accepts Pathname objects' do
     info = pdf.image(Pathname.new(filename))
     expect(info.height).to eq(453)
   end
 
+  context 'closes opened files again after getting Pathnames', issue: 975 do
+    context 'spec with File message spy' do
+      let(:not_filename) { 'non-existent filename' }
+      let(:pathname_double) { instance_double('Pathname', file?: true) }
+
+      before do
+        file_content = File.new(filename, 'rb').read
+        allow(Pathname).to receive(:new).with(not_filename) { pathname_double }
+        allow(pathname_double).to receive(:binread) { file_content }
+      end
+
+      it 'uses only binread, which closes opened files' do
+        # this implicitly tests that a file handle is closed again
+        # because only stubbed binread can be called on not_filename
+        _info = pdf.image(not_filename)
+        expect(pathname_double).to have_received(:binread)
+      end
+    end
+
+    it 'closes opened files, spec with lsof' do
```

Comment:
An easier approach to these kind of specs is to wrap the definition with the platform check, which means you avoid the sometimes-this-spec-is-pending-but-not-always problem:

```ruby
describe "situation" do
  system_has_lsof = system('lsof -v > /dev/null 2>&1')
  system_has_grep = system('grep --version > /dev/null 2>&1')
  if system_has_lsof && system_has_grep
    it 'closes opened files, spec with lsof' do
      ...
    end
  end

  if RUBY_PLATFORM != 'java'
    it 'closes opened files, spec with ObjectSpace' do
      ...
    end
  end
end
```